### PR TITLE
use receiveOne instead of receiveMessageType, #25914

### PR DESCRIPTION
**Pull request by patriknw**
* when the type is the same as the probe's type

Refs #25914

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM I guess

**Comment by johanandren** [comment]
I'll add that I'm not overly enthusiastic about that new probe method, but since I was out of the loop and didn't object when it got in I guess that's just how it is now.

**Review by chbatey — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
> I'll add that I'm not overly enthusiastic about that new probe method

What is not good? When have `receiveN` and I see `receiveOne` as a good complement to `expectMessage` when the assert can't be done with `equals`. It also means that we don't need `expectMessagePF`, which I never liked for several reasons.

**Comment by johanandren** [comment]
To me `expect` communicates more clearly that it's a receive or fail, while receive happens if it happens but not happening isn't really a failure. I also think the explicitness in `expectMessageType[T]` is nicer as it documents exactly what message we are expecting without having to backtrack to what type the probe has, and allows catching the result with the right type etc.

**Comment by patriknw** [comment]
> To me expect communicates more clearly that it's a receive or fail, while receive happens if it happens but not happening isn't really a failure.

I think the difference is that `expect` performs the assert while `receive` only retrieves the message from the probe and then it's up to the test to assert further or use it in other ways.

> I also think the explicitness in expectMessageType[T] is nicer

It's massive amount of repetition of type parameters as can be seen in this PR. That is like saying that one should annotate types in all method calls.

**Comment by johanandren** [comment]
> It's massive amount of repetition of type parameters as can be seen in this PR. That is like saying that one should annotate types in all method calls.

Or in public APIs. Anyways, I'm not ready to go to war over this, just wanted to state that I dislike it. So that it is on record when the judgement day comes.

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala [hunk 1]
```diff
@@ -575,7 +575,7 @@ abstract class ActorContextSpec extends ScalaTestWithActorTestKit(
       }.decorate)
       val adapterName = "hello"
       actor ! adapterName
-      val adapter = probe.expectMessageType[ActorRef[String]]
+      val adapter = probe.receiveOne()
```
Comment (helena) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorContextSpec.scala]: 👍 These changes to the new receiveOne are great, much cleaner, thanks for doing it.

**Review by helena — APPROVED** [review]
LGTM, thanks for doing it

**Review by raboof — COMMENTED** [review]
While I agree `val foo = expectMessageType[Foo]` is overly verbose when `Foo` is already the type associated with the `ActorRef`, I also agree `receiveOne` looks a bit foreign (I haven't seen `receiveN` used much).

Perhaps it would be neat to have a parameterless `def receiveMessage: M` that expects and returns one message of the type associated with the `ActorRef`?

**Comment by ktoso** [comment]
Huge +1 to @raboof

**Comment by patriknw** [comment]
> neat to have a parameterless def receiveMessage: M

so you are suggesting that I rename `receiveOne` to `receiveMessage`? I'm fine with that.

**Comment by patriknw** [comment]
Renamed receiveOne to receiveMessage.
Renamed receiveN to receiveMessages.

**Review by helena — APPROVED** [review]
LGTM

**Pull request merged**